### ISSUE TITLE
FFMPEG path does not update without e.g. clicking the "Capture" button afterwards

### ISSUE THREAD

**Issue by JohnGrime** [Slicer/Slicer#6945]
## Summary

Hi!

I'm trying to make animations with SlicerMorph, but I had problems with updating the `ffmpeg` path in the Screen Capture module.

Changing the path alone does not seem to affect where SlicerMorph's animation export tools look for the `ffmpeg` executable.

However, if I click on the Capture button in the Screen Capture module first, the updated path is then used in the SlicerMorph animation system when exporting a video. I guess the actual path update must be triggered by some connected event, such as clicking on the Capture button pulling the appropriate path string the associated text box and updating the system variables.

## Steps to reproduce

Change the file path to `ffmpeg` in Screen Capture, and try to export an animation in SlicerMorph's animation tools. Note the path actually used in Slicer's message logs.

## Expected behavior

I'd expect the video export to use the path to `ffmpeg` currently specified in the Screen Capture module, as per the [guidance ](https://slicer.readthedocs.io/en/latest/user_guide/modules/screencapture.html#setting-up-ffmpeg).

## Environment
- Slicer version: Slicer-5.2.2 r31382 / fb46bd1
- Windows 10 Enterprise 21H2

**Comment by pieper** [comment]
For reference the ScreenCaptureLogic is used here:
https://github.com/SlicerMorph/SlicerMorph/blob/master/Animator/Animator.py#L933-L949

**Comment by pieper** [comment]
It makes sense that you need to click the Capture button since the path is not set and stored in the registry until then.

https://github.com/Slicer/Slicer/blob/main/Modules/Scripted/ScreenCapture/ScreenCapture.py#L639

https://github.com/Slicer/Slicer/blob/main/Modules/Scripted/ScreenCapture/ScreenCapture.py#L964-L970

Maybe the path should be saved to settings every time the `currentPathChanged` signal is invoked.

@lassoan

**Comment by lassoan** [comment]
Saving current GUI choices into application settings when the user initiates capturing offers some protection against unintended changes (if you messed up something then you can still just shut down the application and nothing is lost/overwritten).

We never update settings on "currentPathChanged" and I don't think we should start it now. Even on focus change would be too implicit. We could add a "validate" button that would run the executable with some simple test (e.g., print version) and save the new path into settings if successful would be acceptable, but that would be yet another button on an already really busy GUI.

If Animator extension needs ffmpeg then it could call ScreenCapture module logic to get/set/detect the ffmpeg path.

**Comment by pieper** [comment]
The Animator should not know about the ScreenCapture GUI and calling `getFfmpegPath` wouldn't fix the problem since that method reads from settings.  In fact the Animator doesn't know anything about ffmpeg at all, it just uses the ScreenCaptureLogic.  

https://github.com/Slicer/Slicer/blob/main/Modules/Scripted/ScreenCapture/ScreenCapture.py#L958-L962


So yes, a Validate or Save button would make sense next to the ctkPathLineEdit.  Or make the path be just a label with an explicit Edit button.

**Comment by JohnGrime** [comment]
The user guide could also be tweaked to recommend clicking the "Capture" button after modifying the `ffmpeg` path to ensure any changes percolate through the system.

That seems like the easiest way to avoid users tripping over this issue, without needing to actually change how anything works?

**Comment by pieper** [comment]
Yes, updating the documentation would be the easiest.  But since you stumbled on this issue in real life I'm guessing other people might as well so fixing the underlying issue would be best if we can.   Unfortunately I can't justify spending time on it right now.
